presto-native-execution/presto_cpp/main/QueryContextManager.h

```diff
@@ -127,6 +128,7 @@ class QueryContextManager {
 
   folly::Executor* const driverExecutor_{nullptr};
   folly::Executor* const spillerExecutor_{nullptr};
+  const std::shared_ptr<SessionProperties> sessionProperties_;
```

Comment:
This can simply be a SessionProperties object. A shared_ptr is not required. The lifetime of SessionProperties is the same as that of a QueryContextManager.